common/src/main/java/com/wynntils/core/config/ConfigManager.java

```diff
@@ -121,14 +121,33 @@ private static List<ConfigHolder> collectConfigOptions(Feature feature) {
         if (featureInfo == null || featureInfo.category().isBlank()) return null;
 
         String category = featureInfo.category();
-        List<ConfigHolder> options = new ArrayList<>();
 
-        // collect options
-        for (Field f : FieldUtils.getFieldsWithAnnotation(feature.getClass(), Config.class)) {
-            Config metadata = f.getAnnotation(Config.class);
-            options.add(new ConfigHolder(feature, f, category, metadata));
+        loadFeatureOverlayConfigOptions(feature, category);
+
+        return getConfigOptions(category, feature);
+    }
+
+    private static void loadFeatureOverlayConfigOptions(Feature feature, String category) {
+        // collect feature's overlays' config options
+        for (Overlay overlay : feature.getOverlays()) {
+            List<ConfigHolder> options = getConfigOptions(category, overlay);
+
+            CONFIG_HOLDERS.addAll(options);
```

Comment:
These actions are shared with `registerFeature`. All must be done to correctly register options. It should be extracted to a common method and called from both here and `registerFeature`.